p2p/src/serde.rs

```diff
@@ -0,0 +1,176 @@
+// Copyright 2020-2021 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::{
+    behaviour::{AddressInfo, PeerAddress},
+    firewall::{FirewallConfiguration, FirewallRules, Rule},
+};
+
+use libp2p::core::{multihash, PeerId};
+use serde::{Deserialize, Serialize};
+use smallvec::SmallVec;
+use std::{
+    collections::HashMap,
+    convert::{TryFrom, TryInto},
+    marker::PhantomData,
+};
+
+#[derive(Serialize, Deserialize, Eq, Hash, Ord, PartialEq, PartialOrd)]
+pub struct SerdePeerId(Vec<u8>);
```

Comment:
Discussion: 
You use a wrapper type to simplify de/serialization of inner types like `Peerid`, `AddressInfo` etc. Do you think it might reasonably reduce the amount of wrapper types, but using a `DeSerializeFirewallConfig` enum type? The variants could capture all necessary data enabling de/serialization with `serde`; providing a `From` implementation would bind to all relevant types.

Reply:
If `PeerId` would implement `Serialize` and `Deserialize`, we could remove most of these workaround types. I have opened an Issue at libp2p https://github.com/libp2p/rust-libp2p/issues/2341 and will work on a PR to maybe solve this.

Reply:
I would merge this for now how it is, and resolve this once the changes will (hopefully) added in libp2p. Okay @felsweg-iota?

Reply:
yes please